generator/src/main/java/com/algorand/sdkutils/listeners/GoGenerator.java

```diff
@@ -0,0 +1,678 @@
+import com.algorand.sdkutils.utils.Tools;
+import com.algorand.sdkutils.utils.TypeDef;
+
+public class GoGenerator extends Subscriber {
+
+    enum Annotation {
+        JSON,
+        CODEC,
+        URL
+    }
+    
+    static final String TAB = "\t";
+
+    // packageName is also the folder where the files sit
+    private String packageName;
+    // filesFolder is rootFolder/packageName
+    private String filesFolder;
+
+    // ModelWriter is a manager of the model file writer
+    // It is limited to only one file at a time.
+    private ModelWriter modelWriter;
+    
+    // Query files
+
+    // This is the individual file for each path query
+    private BufferedWriter queryWriter;
+
+    // path parameters are collected and stored here
+    // this is useful when all the path parameters should be available for 
+    // constructing some functions.  
+    private TreeMap<String, String> pathParameters;
+
+    // queryFunctions hold all the query functions which go to one file 
+    // written by queryWriter
+    private StringBuffer queryFunctions;
+
+    private String currentQueryName;
+    private String currentQueryReturn;
+
+    // imports that go to one query file written by queryWriter
+    private Map<String, Set<String>> imports;
+
+    // pathDesc has the comments and the path string template
+    private String pathDesc;
+    // path encoded with place-holders from the spec
+    private String path;
+
+    // client functions
+
+    // clientFunctions hold all the functions that return the  
+    // makers of the query items. It is to organize them alphabetically sorted
+    // this is reset once at the beginning, and flushed at terminate. 
+    private TreeMap<String, String> clientFunctions;
+
+    // clientFunction holds a single client function as it is getting contructed
+    // If is reset at each new query 
+    private StringBuffer clientFunction;
+
+
+    public GoGenerator(String rootFolder, String packageName, Publisher publisher) throws IOException {
+        publisher.subscribe(Events.NEW_MODEL, this);
+        publisher.subscribe(Events.NEW_PROPERTY, this);
+        publisher.subscribe(Events.NEW_RETURN_TYPE, this);
+        publisher.subscribe(Events.NEW_QUERY, this);
+        publisher.subscribe(Events.QUERY_PARAMETER, this);
+        publisher.subscribe(Events.PATH_PARAMETER, this);
+        publisher.subscribe(Events.BODY_CONTENT, this);
+        publisher.subscribe(Events.END_QUERY, this);
+
+        modelWriter = null;
+        clientFunctions = new TreeMap<String, String>();
+        filesFolder = rootFolder + File.separatorChar + packageName;
+        modelWriter = new ModelWriter(this, filesFolder);
+        this.packageName = packageName;
+    }
+
+    public void terminate() {
+        modelWriter.close();
+        modelWriter = null;
+
+        writeClientFunctions();
+    }
+
+    private void writeClientFunctions() {
+        BufferedWriter bw = newFile("applicationclient", filesFolder);
```

Comment:
Should applicationclient be configurable?

Reply:
Looks like there are several missing configurations here if we want to use this for the indexer and algod client. If we're doing this, we need to be sure to use the generator for both.

Reply:
Yes, you are correct. Next time around we need to generate those from the generator, we should do it. 
This is a shortcut in this PR, since we did not need more for this task.